azuread/configuration.go

```diff
@@ -136,18 +155,129 @@ func (p *azureFedAuthConfig) validateParameters(params map[string]string) error
 		p.adalWorkflow = mssql.FedAuthADALWorkflowNone
 		p.password = params["password"]
 
-		if p.password == "" {
-			return errors.New("Must provide 'password' parameter when using ActiveDirectoryServicePrincipalAccessToken authentication")
+		if p.password == "" {
+			return errors.New("Must provide 'password' parameter when using ActiveDirectoryServicePrincipalAccessToken authentication")
+		}
+	case strings.EqualFold(fedAuthWorkflow, ActiveDirectoryAzurePipelines):
+		p.adalWorkflow = mssql.FedAuthADALWorkflowPassword
+		// Split the clientID@tenantID format
+		p.clientID, p.tenantID = splitTenantAndClientID(params["user id"])
+		if p.clientID == "" {
+			return errors.New("Must provide 'client id[@tenant id]' as username parameter when using ActiveDirectoryAzurePipelines authentication")
+		}
+		p.serviceConnectionID = params["serviceconnectionid"]
+		p.systemAccessToken = params["systemtoken"]
+		if p.serviceConnectionID == "" {
+			return errors.New("Must provide 'serviceconnectionid' parameter when using ActiveDirectoryAzurePipelines authentication")
+		}
+		if p.systemAccessToken == "" {
```

Comment:
>systemAccessTo [](http://example.com/codeflow?start=7&length=14)

 The environment variables AZURESUBSCRIPTION_CLIENT_ID, AZURESUBSCRIPTION_TENANT_ID, and AZURESUBSCRIPTION_SERVICE_CONNECTION_ID are configured by Azure Pipelines only in the tasks AzureCLI@2 and AzurePowerShell@5. 

If the connection string doesn't provide these values, let's get their values from their respective environment variables here, and only throw if they aren't set in either the connection string or the environment. Update the README to mention this case.